pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go

```diff
@@ -192,37 +192,26 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
 		})
 	})
 
-	DescribeTable("should apply defaults on VMI create", func(arch string, cpuModel string) {
+	DescribeTable("should apply defaults on VMI create", func(arch string, cpuModel string, machineType string) {
 		// no limits wanted on this test, to not copy the limit to requests
 
-		if arch == "" {
-			if rt.GOARCH == "arm64" {
-				cpuModel = v1.CPUModeHostPassthrough
-			} else {
-				cpuModel = v1.DefaultCPUModel
-			}
+		if machineType == "" {
+			machineType = mutator.ClusterConfig.GetMachineType(rt.GOARCH)
 		}
-		_, vmiSpec, _ := getMetaSpecStatusFromAdmit(arch)
 
-		if webhooks.IsPPC64(vmiSpec) {
-			Expect(vmiSpec.Domain.Machine.Type).To(Equal("pseries"))
-		} else if webhooks.IsARM64(vmiSpec) {
-			Expect(vmiSpec.Domain.Machine.Type).To(Equal("virt"))
-		} else if webhooks.IsS390X(vmiSpec) {
-			Expect(vmiSpec.Domain.Machine.Type).To(Equal("s390-ccw-virtio"))
-		} else {
-			Expect(vmiSpec.Domain.Machine.Type).To(Equal("q35"))
-		}
+		_, vmiSpec, _ := getMetaSpecStatusFromAdmit(arch)
 
+		Expect(vmiSpec.Domain.Machine.Type).To(Equal(machineType))
 		Expect(vmiSpec.Domain.CPU.Model).To(Equal(cpuModel))
 		Expect(vmiSpec.Domain.Resources.Requests.Cpu().IsZero()).To(BeTrue())
 		Expect(vmiSpec.Domain.Resources.Requests.Memory().Value()).To(Equal(int64(0)))
 	},
-		Entry("when architecture is amd64", "amd64", v1.DefaultCPUModel),
-		Entry("when architecture is arm64", "arm64", v1.CPUModeHostPassthrough),
-		Entry("when architecture is ppc64le", "ppc64le", v1.DefaultCPUModel),
-		Entry("when architecture is s390x", "s390x", v1.DefaultCPUModel),
-		Entry("when architecture is not specified", "", v1.DefaultCPUModel))
+		Entry("when architecture is amd64", "amd64", v1.DefaultCPUModel, "q35"),
+		Entry("when architecture is arm64", "arm64", v1.CPUModeHostPassthrough, "virt"),
+		Entry("when architecture is ppc64le", "ppc64le", v1.DefaultCPUModel, "pseries"),
+		Entry("when architecture is s390x", "s390x", v1.DefaultCPUModel, "s390-ccw-virtio"),
+		Entry("when architecture is not specified", "", v1.DefaultCPUModel, ""),
```

Comment:
Won't that break on arm since it is using `v1.DefaultCPUModel` instead of `v1.CPUModeHostPassthrough` now? Ideally our tests would not depend on the host architecture at all. I don't see which value this last test case has?

Reply:
Right, i forgot about arm...

As for the value, it checks that we can deal with an unknown Architecture without error. I think that is the best way to test it, since most VMs will likely not have the arch or machine type set and rely on this code path.

Reply:
Wait we are not running any arm64 unit tests in the CI? I don't see a test lane below for that?

Reply:
Can you extract the last test case into another table where you set the default architecture in the cluster configuration, so you can predict what should happen? This way we can get rid of the `rt.GOARCH` dependency and test all possible arch combinations independent of the architecture of the host executing the tests (which IMO is pretty bad). It could be in a follow up.

Reply:
Yes, i can do that as a follow up.